examples/advanced/xgboost/fedxgb/README.md

```diff
@@ -0,0 +1,203 @@
+
+This scheme bears certain similarity to the [Random Forest mode](https://xgboost.readthedocs.io/en/stable/tutorials/rf.html) of XGBoost, where a `num_parallel_tree` is boosted based on random row/col splits, rather than a single tree. Under federated learning setting, such split is fixed to clients rather than random and without column subsampling. 
+
+In addition to basic uniform shrinkage setting where all clients have the same learning rate, based on our research, we enabled scaled shrinkage across clients for weighted aggregation according to each client's data size, which is shown to significantly improve the model's performance on non-uniform quantity splits over HIGGS data.
+
+
+Specifically, the global model is updated by aggregating the trees from all clients as a forest, and the global model is then broadcasted back to all clients for local prediction and further training.
+
+The XGBoost Booster api is leveraged to create in-memory Booster objects that persist across rounds to cache predictions from trees added in previous rounds and retain other data structures needed for training.
+
+## Vertical Federated XGBoost
+Under vertical setting, each participant joining the federated learning will 
+have part of the whole features, while each site has all the overlapping instances.
+
+### Private Set Intersection (PSI)
+Since not every site will have the same set of data samples (rows), we will use PSI to first compare encrypted versions of the sites' datasets in order to jointly compute the intersection based on common IDs. In the following example, we add a `uid_{idx}` to each instance and give each site 
+a portion of the dataset that includes a common overlap. After PSI, the identifiers are dropped since they are only used for matching, and training is then done on the intersected data. To learn more about our PSI protocol implementation, see our [psi example](../../psi/README.md).
+> **_NOTE:_** The uid can be a composition of multiple variables with a transformation, however in this example we use indices for simplicity. PSI can also be used for computing the intersection of overlapping features, but here we give each site unique features.
+
+### Histogram-based Collaboration
+Similar to its horizontal counterpart, under vertical collaboration, histogram-based collaboration will 
+aggregate the gradient information from each site and update the global model accordingly, resulting in
+the same model as the centralized / histogram-based horizontal training. 
+We leverage the [vertical federated learning support](https://github.com/dmlc/xgboost/issues/8424) in the XGBoost open-source library. This allows for the distributed XGBoost algorithm to operate in a federated manner on vertically split data.
+
+## Data Preparation
+Assuming the HIGGS data has been downloaded following [the instructions](../README.md), we further split the data 
+horizontally and vertically for federated learning.
```

Comment:
Could we say something about "reality" here, then talk about how we simulate the reality?

For example: "In vertical settings, each party holds different features of the same data samples. To simulate this, we split the HIGGS data by columns. In horizontal settings, each party holds different data samples with the same set of features. To simulate this, we split the HIGGS data by rows."

Reply:
added